domain/objectstore/state/state.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+
+	coredatabase "github.com/juju/juju/core/database"
+	"github.com/juju/juju/domain"
+	"github.com/juju/juju/domain/objectstore"
+)
+
+// State implements the domain objectstore state.
+type State struct {
+	*domain.StateBase
+}
+
+// NewState returns a new State instance.
+func NewState(factory coredatabase.TxnRunnerFactory) *State {
+	return &State{
+		StateBase: domain.NewStateBase(factory),
+	}
+}
+
+// GetMetadata returns the persistence metadata for the specified path.
+func (s *State) GetMetadata(ctx context.Context, path string) (objectstore.Metadata, error) {
+	db, err := s.DB()
+	if err != nil {
+		return objectstore.Metadata{}, err
+	}
+
+	query := `
+SELECT path, metadata_uuid,  object_store_metadata.size, object_store_metadata.hash
```

Comment:
Odd double spaces here, and the text to the right of the left-most keywords should be aligned.